README.md

```diff
@@ -245,6 +255,24 @@ The second `as` parameter for `push` and `replace` is an optional _decoration_ o
 
 _Note: in order to programmatically change the route without triggering navigation and component-fetching, use `props.url.push` and `props.url.replace` within a component_
 
+##### Router Events
+
+You can also listen to different events happening inside the Router. For that, `next/router` comes with a Node.js [EventEmitter](https://nodejs.org/api/events.html) compatible API.
+
+Supported events:
+
+- `routeChangeStart(route, url)` - Fires when a route starts to change
+- `routeChangeComplete(route, url)` - Fires when a route changed completely
+- `routeChangeError(err, route, url)` - Fires when there's an error when changing routes
+
+Here's how to property listen to a router event:
+
+```js
+Router.on('routeChangeStart', (route, url) => {
+  console.log('Route is changing to: ', route)
+})
```

Comment:
How about removing that listener?
I choose eventEmitter because of that. (It's familiar)

And there are some other events as well.
But we could return  a stop function as well.